marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/prov.sjs

```diff
@@ -241,7 +241,7 @@ class Provenance {
    *      - location (doc URI)
    */
   _createIngestionStepRecord(jobId, flowId, stepName, stepDefinitionName, docURI, info) {
-    let provId = `${jobId + flowId + 'ingestion' + docURI}`;
+    let provId = `${jobId + flowId + 'ingestion' + docURI}_${sem.uuidString()}`;
```

Comment:
Per the write-up in 6413, I think all we need is a UUID. The jobId itself is a UUID, so we now have two UUIDs in this. Also see the "How provenance works" Wiki page for details on how this is constructed. 

There's also a lot of duplicated code in this library; we don't want to continue that pattern, we want to clean that up. 

So add a "newProvId()" function that each of these functions uses, and that can just return a UUID.

Reply:
The provenanceId being a UUID will be addressed in a separate ticket

Reply:
Let's still get a "newProvId(jobId, flowId, stepType, docUri)" function in place - that avoids duplicating the logic of how the prov ID is constructed in many different places.